File: 007/007.rb
limit = 10001
counter = 0
x=z=2

prime = true
loop do
  while z < x && prime do
    if x % z == 0
      prime = false
      break
    end
    z += 1
  end
  z = 2
  if prime
    counter += 1
    puts counter    
    if counter == limit
      print x
      break
    end
  end
  prime = true
  x += 1
end

File: 007/007.c
int main(void){
int x,z,counter,limit,prime;
limit = 10001;
counter = 0;
x=z=2;
prime = 1;
	while(x > 0){
		z = 2;
		while(z < x){
		
			if( x % z == 0){
				prime = 0;
				break;
			}
			z++;
		}
		if(prime == 1){
			counter++;
		}
		
		if(counter == limit){
				break;
		}
	
		prime = 1;
		x++;
	}
	printf("%d", x);
	return 0;
}